WordPress/gutenberg
# Run PHPUnit jobs across multiple PHP versions

## What?
This PR:
- Removes PHP related checks from the `Unit Tests` Github CI job;
- Adds 2 new Github CI Jobs: `PHPUnit Tests` and `PHP Coding Standards` (the same way it's implemented in Core);
- **brings ability to run PHPUnit tests on all PHP versions (5.6 through 8.2) using the CI jobs;**
- shows code style errors on the "Files changed" tab (same as Core);
- Fixes an issue with `wp-env` as it used non-optimal PHPUnit versions on PHP 7.3 and 7.4 (props to @jrfnl);

## Why?
Gutenberg should support the same PHP versions as Core.

## How?
This PR adds 2 new Github CI Jobs: `PHPUnit Tests` and `PHP Coding Standards`.
These PHP CI jobs were backported from Core to ensure that we test Gutenberg PRs in the same fashion.

## Testing Instructions
1. Make sure that the CI jobs run without errors.
2. Observe there are `PHP_VERSION (multisite) on ubuntu-latest` jobs for each supported PHP version (5.6 through 8.2).

Props @jrfnl , @dmsnell , @anton-vlasenko
